api/plugins/session-auth.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  DoneFuncWithErrOrRes,
+  FastifyPluginCallback,
+  FastifyReply,
+  FastifyRequest
+} from 'fastify';
+import fp from 'fastify-plugin';
+
+const sessionAuth: FastifyPluginCallback = (fastify, _opts, done) => {
+  fastify.decorate('authenticateSession', authenticateSession);
+
+  function authenticateSession(
+    req: FastifyRequest,
+    res: FastifyReply,
+    done: DoneFuncWithErrOrRes
+  ) {
+    if (req.session.user === undefined) {
```

Comment:
```suggestion
    if (!req.session.user) {
```
It's a little more concise and I think more correct: any falsy value should be rejected.